Source/DafnyCore/AST/Members/Function.cs

```diff
@@ -507,6 +507,12 @@ public string GetTriviaContainingDocstring() {
       return tentativeTrivia;
     }
 
+    // Comments after the type, but before the clauses
+    tentativeTrivia = (ResultType.EndToken.TrailingTrivia + ResultType.EndToken.Next?.LeadingTrivia).Trim();
```

Comment:
Without this code, the tests `DocStringForFunctions` and `DocstringWorksForFunctions` fail. They no longer detect the docstring in:

```
  function Test5(): int // [START Docstring5
    // END Docstring5]
  {
    1
  }
```

And looked and the code and was unable to figure out how it was able to find that in the first place, since `int` nor `{` is owned by `Function`, and it seems `Function.GetTriviaContainingDocstring` just traverses tokens owned by Function. I added this code to make those tests pass again.

Reply:
I used master to debug this test case to figure out what happens and the following:

```
var endTokenDefinition =
      OwnedTokens.LastOrDefault(token => token.val == ")" || token.pos == ResultType.EndToken.pos);
```
was returning "int", so I think "int" was owned by the function.
This means that in your PR, somehow "int" is no longer owned by the function. 

I don't understand from your PR how the ownership of "int" disappeared from the function (I would appreciate if you could explain why), but anyway, I think the following would be better suited as a fix.

```
var endTokenDefinition =
      OwnedTokens.LastOrDefault(token => token.val == ")");
if (endTokenDefinition != null && endTokenDefinition.pos < ResultType.EndToken.pos) {
  endTokenDefinition  = ResultType.EndToken
}
```

The reason for that is that we will not consider lines after the function as a docstring if the type was the last token of the function, and the logic above takes care of that particular case.

Reply:
> was returning "int", so I think "int" was owned by the function.
This means that in your PR, somehow "int" is no longer owned by the function.

> I don't understand from your PR how the ownership of "int" disappeared from the function (I would appreciate if you could explain why), but anyway, I think the following would be better suited as a fix.

Thanks for the deep dive. To be honest, I imagined that change of ownership was the reason for the change in behavior, but given that the return type is its own AST node, I think not having the function own the return type token is the correct behavior, so I just adapted the code to work with that. I think previously the "range" of the return type AST node was set incorrectly, causing it not to find its token when determining ownership.

Applied your suggested change. Thanks